src/storage/v3/name_id_mapper.hpp

```diff
@@ -11,88 +11,42 @@
-    auto name_to_id_acc = name_to_id_.access();
-    auto found = name_to_id_acc.find(name);
-    uint64_t id{0};
-    if (found == name_to_id_acc.end()) {
-      uint64_t new_id = counter_.fetch_add(1, std::memory_order_acq_rel);
-      // Try to insert the mapping with the `new_id`, but use the id that is in
-      // the object itself. The object that cointains the mapping is designed to
-      // be a map, so that if the inserted name already exists `insert` will
-      // return an iterator to the existing item. This prevents assignment of
-      // two IDs to the same name when the mapping is being inserted
-      // concurrently from two threads. One ID is wasted in that case, though.
-      id = name_to_id_acc.insert({std::string(name), new_id}).first->id;
-    } else {
-      id = found->id;
-    }
-    auto id_to_name_acc = id_to_name_.access();
-    // We have to try to insert the ID to name mapping even if we are not the
-    // one who assigned the ID because we have to make sure that after this
-    // method returns that both mappings exist.
-    if (id_to_name_acc.find(id) == id_to_name_acc.end()) {
-      // We first try to find the `id` in the map to avoid making an unnecessary
-      // temporary memory allocation when the object already exists.
-      id_to_name_acc.insert({id, std::string(name)});
+  uint64_t NameToId(std::string_view name) const {
+    auto it =
+        std::ranges::find_if(id_to_name_, [name](const auto &name_id_pair) { return name_id_pair.second == name; });
+    if (it == id_to_name_.end()) {
+      return kUnmappedId;
     }
-    return id;
+    return it->first;
   }
 
-  // NOTE: Currently this function returns a `const std::string &` instead of a
-  // `std::string` to avoid making unnecessary copies of the string.
-  // Usually, this wouldn't be correct because the accessor to the
-  // `utils::SkipList` is destroyed in this function and that removes the
-  // guarantee that the reference to the value contained in the list will be
-  // valid.
-  // Currently, we never delete anything from the `utils::SkipList` so the
-  // references will always be valid. If you change this class to remove unused
-  // names, be sure to change the signature of this function.
-  const std::string &IdToName(uint64_t id) const {
-    auto id_to_name_acc = id_to_name_.access();
-    auto result = id_to_name_acc.find(id);
-    MG_ASSERT(result != id_to_name_acc.end(), "Trying to get a name for an invalid ID!");
-    return result->name;
+  const std::string &IdToName(const uint64_t id) const {
+    auto it = std::ranges::find_if(id_to_name_, [id](const auto &name_id_pair) { return name_id_pair.first == id; });
+    MG_ASSERT(it != id_to_name_.end(), "Id not know in mapper!");
+    return it->second;
   }
 
  private:
-  std::atomic<uint64_t> counter_{0};
-  utils::SkipList<MapNameToId> name_to_id_;
-  utils::SkipList<MapIdToName> id_to_name_;
+  std::unordered_map<uint64_t, std::string> id_to_name_;
```

Comment:
I think we have to introduce another map to store the mapping in the name -> id direction to avoid linear lookups.